### PR TITLE
ios-study-56,ios-study-58,ios-study-60,Algorithm: arithmetic progression,Algorithm: geometric progression,Algorithm: sqrt & степінь

### PR DESCRIPTION
**Pull request by oleksandr-syurpita**
## What's New

1. Algorithm: arithmetic progression 

- Task 1 complete
- Task 2 complete
- Task 3 complete
- Task 4 complete
- Task 5 complete
2. Algorithm: geometric progression

- Task 1 complete
- Task 2 complete
- Task 3 complete
- Task 4 complete
- Task 5 complete

3. Algorithm: sqrt & степінь

- Task 1 complete
- Task 2 complete
- Task 3 complete
- Task 4 complete
- Task 5 complete
- Task 6 complete
- Task 7 complete
- Task 8 complete
- Task 9 not complete
- Task 10 complete

## Issue

- ios-study-56
- ios-study-58
- ios-study-60,

### CODE REVIEW

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
+
+//: [Previous](@previous)
+
+import Foundation
+
+
+//Calculate sum: 1 + 2 + 3 + 4 + 5 + 6 + 7+ ..., where n is count of numbers
+func functionTask3_1(i: Int) ->Int {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:8]: n should be the parameter

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    guard i > 0 else { return 0 }
+    
+    var loop = 0
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:11]: better use `i` or `iteration`

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    var sum = 0
+    var number = 0
+    
+    while i > loop {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:15]: better use `i < n`

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        number += 1
+        sum += number
+        loop += 1
+    }
+    return sum
+
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:21]: remove extra line

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    
+    while i > loop {
+        number += 1
+        sum += number
+        loop += 1
+    }
+    return sum
+    
+}
+functionTask3_1(i: 3)
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:23]:
```swift
let result3_1 = functionTask3_1(i: 3)
print("Result for 3.1 - \(result3_1)" )
```
do the same for the rest of the functions

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+print("##################")
+func functionTask_3_2(n: Int, i: Int) -> Int {
+    
+    //Calculate sum: 1 + 3 + 5 + 7 + ..., where n is count of numbers
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:28]: - move comment with the task before the function (line 25) 
- do the same for all the rest functions

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        sum += number
+        loop += 1
+    }
+    return sum
+    
+}
+functionTask3_1(i: 3)
+
+print("##################")
+func functionTask_3_2(n: Int, i: Int) -> Int {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:26]: why does it accept two parameters ?
as I understand it should accept only `n: Int`

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    
+    //Calculate sum: 1 + 3 + 5 + 7 + ..., where n is count of numbers
+    guard n > 0 else { return 0 }
+    guard i > 0 else { return 0 }
+    var loop = 0
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:31]: - use `i` instead
- do the same for all the rest

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    var sum = 0
+    var  number = 1
+    
+    while i > loop {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:35]: - `i < n`
- do the same for all the rest

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        sum +=  number
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:36]: extra space

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        number += n
+        loop += 1
+    }
+
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:40]: remove empty line

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+print("%%%%%%%%%%")
+func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int {
+    //Write function: `func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int`, where start is the start number, step is the difference between two near numbers, n is count of numbers. Function should return its sum.
+    var loop = 0
+    var sum = 0
+    var  number = start
+    
+    while n > loop {
+        sum +=  number
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:88]: extra space

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+functionTask_3_4(n: 3)
+
+print("%%%%%%%%%%")
+func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int {
+    //Write function: `func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int`, where start is the start number, step is the difference between two near numbers, n is count of numbers. Function should return its sum.
+    var loop = 0
+    var sum = 0
+    var  number = start
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:85]: extra space

File: RomanTask.playground/Pages/Algorithm_geometric.xcplaygroundpage/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,94 @@
+//: [Previous](@previous)
+
+import Foundation
+
+print("^^^^^^^^^^^")
+func functionTask_4_1(n: Int) -> Int {
+    //Calculate sum: 1 + 2 + 4 + 8 + 16 + 32 + ..., where n is count of numbers
+
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_geometric.xcplaygroundpage/Contents.swift:8]: remove extra lines

**Review by romansorochak — CHANGES_REQUESTED** [review]
Approved
Though lots of small mistakes in style guide.
Fix them all and ask for review again

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift (hunk 1 above)
Comment (oleksandr-syurpita) [reply RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:8]: done

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift (hunk 2 above)
Comment (oleksandr-syurpita) [reply RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:11]: done

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift (hunk 3 above)
Comment (oleksandr-syurpita) [reply RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:15]: done

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+//Calculate sum: 1 + 3 + 5 + 7 + ..., where n is count of numbers
+func functionTask_3_2(n: Int, i: Int) -> Int {
+    guard n > 0 else { return 0 }
+    guard i > 0 else { return 0 }
+    var iteration = 0
+    var sum = 0
+    var number = 1
+    
+    while iteration < i {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:31]: - iteration < n - read the task again
- do the same for all the same moments

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+let result3_2 = functionTask_3_2(n: 4)
+print("Result for 3.2 - \(result3_2)" )
+
+//    Calculate sum: 5 - 4 - 3 - 2 - 1 - 0 - (-1) - (-2) - ... , where n is count of numbers
+func functionTask_3_3(n: Int, i: Int) -> Int {
+    var iteration = 1
+    var number = n
+    var sum = number
+    
+    while iteration < i {
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:46]: `iteration < n`
de facto `i` is used for iteration
it is not obvious to use it as count. 
Instead you should use `n` here. That was also mandatory based on task description.

File: RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    return sum
+}
+let result3_4 = functionTask_3_4(n: 3)
+print("Result for 3.4 - \(result3_4)" )
+
+//Write function: `func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int`, where start is the start number, step is the difference between two near numbers, n is count of numbers. Function should return its sum.
+func sumArithmeticProgression(start: Int, step: Int, n: Int) -> Int {
+    guard start > 0 else { return 0 }
+    guard step > 0 else { return 0 }
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_arithmetic .xcplaygroundpage/Contents.swift:76]: wrong.
step can be negative. 
4 - 3 - 2 - 1 - 0 ... is actually also an arithmetic progression

File: RomanTask.playground/Pages/Algorithm_geometric.xcplaygroundpage/Contents.swift [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    return result
+}
+let result4_4 = functionTask_4_4(n: 4)
+print("Result for 4.4 - \(result4_4)" )
+
+//    Write function: `func sumGeometricProgression(start: Double, step: Double, n: Int) -> Double`, where start is the start number, step is the difference between two near numbers, n is count of numbers. Function should return its sum.
+func sumGeometricProgression(start: Double, step: Double, n: Int) -> Double {
+    guard start > 0 else { return 0 }
+    guard step > 0 else { return 0 }
```
Comment (romansorochak) [inline RomanTask.playground/Pages/Algorithm_geometric.xcplaygroundpage/Contents.swift:77]: wrong

**Review by romansorochak — CHANGES_REQUESTED** [review]

**Pull request merged**